### PR TITLE
Update jsonnet-libs and use Grafana v7.0 features

### PR DESCRIPTION
**Pull request by jdbaldry**
- Also configure Loki data source to user internal Jaeger trace viewer

@buro9 for visibility.

I've manually tested the changes in this PR from a clean cluster and I personally think the internal trace view provides a nice refinement to the TNS story.

### CODE REVIEW

**Comment by jdbaldry** [comment]
I've avoided Jsonnet formatting fixes in this PR but will happily contribute some linting and formatting if it is desirable. I wonder if we have a common Makefile for this since a number of projects have Jsonnet?

File: production/sample/default/main.jsonnet [hunk 1]
```diff
@@ -45,7 +49,7 @@ prometheus + promtail + tns_mixin + {
   },
 
   _images+:: {
-    grafana: 'grafana/grafana:6.6.1',
+    grafana: 'grafana/grafana:7.0.3',
```
Comment (tomwilkie) [inline production/sample/default/main.jsonnet:52]: Can you do this in grafana/jsonnet-libs instead?  And remove it from here.

Comment (buro9) [reply production/sample/default/main.jsonnet:52]: Ah! This was my change in my local repo (before I handed over the laptop and lost the change). Thanks for the pointer 👍

**Comment by tomwilkie** [comment]
> I've avoided Jsonnet formatting fixes in this PR but will happily contribute some linting and formatting if it is desirable. 

Yes please!

> I wonder if we have a common Makefile for this since a number of projects have Jsonnet?

Generally not found it to be worth the hassle; just copy and paste it from jsonnet-libs.

**Comment by jdbaldry** [comment]
Recent changes to the prometheus-ksonnet library have changed the instance label on scraped jobs (from `<pod>` to `<pod>:<container>:<containerPortName>`) which breaks the pivot in Explore from Prometheus to Loki on the instance label. You can still pivot on the job label. 

To me, it feels like instance should match between Loki and Prometheus but the problem arises that a single container can have multiple instances of Prometheus metrics but only one of logs.

**Comment by tomwilkie** [comment]
Can you work with @beorn7 and @slim-bean to come up with a solution? We recognised this would happen when we changed the prometheus relabelling rules, but didn't follow up.

I think the prometheus relabelling rules are fine now, Id say we should drop the instance label from loki (and use a pod label instead).

**Comment by beorn7** [comment]
Yeah, I think `pod` is the label to go with. Plus `container`, if Loki takes log streams per container. 

cluster/namespace/pod/container is the tuple I'd prefer, generally. It follows the drilldown trajectory.

**Comment by jdbaldry** [comment]
Thanks @beorn7, that makes a lot of sense. I made sure to double check the labels in that tuple for parity between Promtail and Prometheus relabelling and realized that I needed to bump the vendored promtail too due to the container_name -> container changes in k8s 1.16. 

I'm pretty happy with this demo now, you can pivot on all the important labels as mentioned and now we get that sweet internal trace view.

**Pull request merged**